tests/components/modbus/test_modbus.py

```diff
@@ -17,56 +31,512 @@
     CONF_TIMEOUT,
     CONF_TYPE,
 )
+from homeassistant.setup import async_setup_component
+
+from .conftest import ReadResult
 
-from .conftest import base_config_test
+
+async def _config_helper(hass, do_config):
+    """Run test for modbus."""
+
+    with mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusTcpClient"
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusSerialClient"
+    ), mock.patch(
+        "homeassistant.components.modbus.modbus.ModbusUdpClient"
+    ):
+        assert await async_setup_component(hass, DOMAIN, do_config) is True
+        await hass.async_block_till_done()
 
 
-@pytest.mark.parametrize("do_discovery", [False, True])
 @pytest.mark.parametrize(
-    "do_options",
+    "do_config",
     [
-        {},
         {
-            CONF_NAME: "modbusTest",
-            CONF_TIMEOUT: 30,
-            CONF_DELAY: 10,
+            DOMAIN: [
```

Comment:
You can simplify these configs by wrapping `do_config` with `{DOMAIN: do_config}` inside `_config_helper`.

Reply:
Done. The reason I did it, was because one who looked at the code had problems seeing the total configuration.